### PR TITLE
[Chat Color] Add ability to change public chat friend name color in config

### PR DESCRIPTION
**Pull request by bmiddle**
I made this change off of issue #3531, it allows users to change friend usernames when they appear in public chat.

### CODE REVIEW

**Comment by bmiddle** [comment]
I'm not sure if the friend check is optimal, it seemed that using the `client.isFriended` returned true even for the local player so I was required to add an additional conditional.

File: runelite-client/src/main/java/net/runelite/client/config/ChatColorConfig.java [hunk 1]
```diff
@@ -490,4 +498,12 @@ default Color transparentExamineHighlight()
 		description = "Color of Usernames in Clan Chat (transparent)"
 	)
 	Color transparentClanUsernames();
+
+	@ConfigItem(
+			position = 87,
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/config/ChatColorConfig.java]: Indent with one tab.

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 2]
```diff
@@ -122,7 +122,15 @@ public void onSetMessage(SetMessage setMessage)
 			case TRADE:
 			case AUTOCHAT:
 			case PUBLIC:
-				usernameColor = isChatboxTransparent ? chatColorConfig.transparentUsername() : chatColorConfig.opaqueUsername();
+				boolean isFriend = client.isFriended(setMessage.getName(), true) && !client.getLocalPlayer().getName().equals(setMessage.getName());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: Does this normally say you are your own friend? I would guess that is impossible.

Comment (bmiddle) [reply runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: I thought so too, I'll open a ticket so I can explain it in a bit

File: runelite-client/src/main/java/net/runelite/client/config/ChatColorConfig.java (hunk 1 above)
Comment (bmiddle) [reply runelite-client/src/main/java/net/runelite/client/config/ChatColorConfig.java]: These things sometimes slip by I'll fix this in a bit

Comment (bmiddle) [reply runelite-client/src/main/java/net/runelite/client/config/ChatColorConfig.java]: Fixed

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java (hunk 2 above)
Comment (bmiddle) [reply runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: #3694

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 3]
```diff
@@ -122,7 +122,15 @@ public void onSetMessage(SetMessage setMessage)
...
+				if (!isFriend)
+				{
+					usernameColor = isChatboxTransparent ? chatColorConfig.transparentUsername() : chatColorConfig.opaqueUsername();
+				}
+				else
+				{
+					usernameColor = isChatboxTransparent ? chatColorConfig.transparentPublicFriendUsernames() : chatColorConfig.opaquePublicFriendUsernames();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: If this color is not set it should probably fall back to the normal colors?

**Comment by bmiddle** [comment]
How's this, Adam?

**Pull request merged**